pkg/cmd/experimental/diagnostics/config.go

```diff
@@ -23,6 +25,9 @@ func (o DiagnosticsOptions) detectClientConfig() (bool, []types.DiagnosticError,
 func (o DiagnosticsOptions) buildRawConfig() (*clientcmdapi.Config, error) {
 	kubeConfig, configErr := o.Factory.OpenShiftClientConfig.RawConfig()
 	if len(kubeConfig.Contexts) == 0 {
+		if configErr == nil {
```

Comment:
This will be easier to reason about and more consistent if you move this check above the `len() == 0` check.